apps/browser/src/autofill/services/autofill.service.ts

```diff
@@ -742,6 +786,99 @@ export default class AutofillService implements AutofillServiceInterface {
     return fillScript;
   }
 
+  /**
+   * Determines whether to warn the user about filling an iframe
+   * @param pageUrl The url of the page/iframe, usually from AutofillPageDetails
+   * @param tabUrl The url of the tab, usually from the message sender (should not come from a content script because
+   *  that is likely to be incorrect in the case of iframes)
+   * @param loginItem The cipher to be filled
+   * @returns `true` if the iframe is untrusted and the warning should be shown, `false` otherwise
+   */
+  untrustedIframe(pageUrl: string, tabUrl: string, loginItem: CipherView): boolean {
+    const pageUrlDomain = Utils.getDomain(pageUrl);
+
+    // Step 1: we trust the page if the page domain matches the tab domain
+    // This means there's either no iframe or the iframe has the same domain and can be trusted
+    if (pageUrlDomain === Utils.getDomain(tabUrl)) {
+      return false;
+    }
+
+    // Step 2: we trust the page if the page domain matches an equivalent domain of the tab
+    const equivalentDomains = this.settingsService.getEquivalentDomains(tabUrl);
+    if (equivalentDomains?.includes(pageUrlDomain)) {
+      this.logService.debug(
+        "iframe at " + pageUrl + " trusted because it is considered equivalent to " + tabUrl
+      );
+      return false;
+    }
+
+    // Step 3: Check the pageUrl against cipher URIs using the configured match detection.
+    // If we get to this point, it is assumed that the tabUrl already matches,
```

Comment:
```suggestion
    // If we are in this function at all, it is assumed that the tabUrl already matches a URL for `loginItem`,
```

Is this intended to mean "this point" in the function, or "this point" in the overall code flow that caused `untrustedIFrame` to be called at all?

Reply:
The former!